<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>678</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(111, 111, 111);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QMdiArea" name="mdiArea">
      <property name="styleSheet">
       <string notr="true">background-color: rgb(77, 77, 77);</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_N"/>
    <addaction name="action_3"/>
    <addaction name="action_S"/>
    <addaction name="action_H"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="action_Z"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="action_4"/>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>Filter</string>
    </property>
    <addaction name="action_6"/>
    <addaction name="action_7"/>
    <addaction name="actionASCII"/>
    <addaction name="action_Sobel"/>
    <addaction name="action_Canny"/>
    <addaction name="action_8"/>
    <addaction name="action_9"/>
    <addaction name="action_10"/>
    <addaction name="action"/>
   </widget>
   <widget class="QMenu" name="menu_H">
    <property name="title">
     <string>Help(&amp;H)</string>
    </property>
    <addaction name="actionAbout_IMAGEW"/>
    <addaction name="action_L"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>Video</string>
    </property>
    <addaction name="action_2"/>
    <addaction name="action_5"/>
    <addaction name="actionSmooth"/>
    <addaction name="actionMosaic"/>
    <addaction name="actionBinarization"/>
    <addaction name="actionEdge_Detection"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_5"/>
   <addaction name="menu_H"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="minimumSize">
    <size>
     <width>50</width>
     <height>44</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>50</width>
     <height>524287</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(111, 111, 111);</string>
   </property>
   <property name="windowTitle">
    <string>工具箱</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <widget class="QPushButton" name="brightnessButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/亮度.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="grayButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/灰度.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="binarizationButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/二值化.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="gammaButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>330</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/伽马变换.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="borderButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/边缘复合.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="gaussButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/高斯滤波.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="sobelButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>210</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/Sobel.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="cannyButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/Canny.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="asciiButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/ASCII.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="filtrerButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>370</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/均值滤波.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="videoButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>410</y>
       <width>31</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-image: url(:/icon/image/icon/视频.png);
background-color: rgb(77, 77, 77);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(111, 111, 111);</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="openfiles"/>
   <addaction name="openfile"/>
   <addaction name="savafile"/>
   <addaction name="action_ascii"/>
  </widget>
  <widget class="QDockWidget" name="brightdockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>200</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>200</width>
     <height>524287</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 10pt &quot;Elephant&quot;;</string>
   </property>
   <property name="windowTitle">
    <string>Brightness / contrast</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <widget class="QLabel" name="brightlabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#f5f5f5;&quot;&gt;Brightness：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#f0f0f0;&quot;&gt;Constrast：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSlider" name="birghtness">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>160</width>
       <height>19</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-255</number>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="contrast">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>160</width>
       <height>19</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="labelb">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>30</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="labelc">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>100</y>
       <width>54</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="braindockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>100</height>
    </size>
   </property>
   <property name="font">
    <font>
     <family>Elephant</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="windowTitle">
    <string>Binarization</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <widget class="QLabel" name="brightlabel_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 10pt &quot;Elephant&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#f1f1f1;&quot;&gt;Threshold:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSlider" name="nSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>160</width>
       <height>19</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>95</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="labelt">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>20</y>
       <width>54</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="gammadockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>100</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 10pt &quot;Elephant&quot;;</string>
   </property>
   <property name="windowTitle">
    <string>Gamma transformation</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_7">
    <widget class="QLabel" name="brightlabel_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ececec;&quot;&gt;Hardness：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSlider" name="horizontalSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>160</width>
       <height>19</height>
      </rect>
     </property>
     <property name="minimum">
      <number>10</number>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="labelh">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>54</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="filterdockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>100</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 10pt &quot;Elephant&quot;;</string>
   </property>
   <property name="windowTitle">
    <string>Mean filtering</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_8">
    <widget class="QLabel" name="brightlabel_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#f1f1f1;&quot;&gt;Hardness：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QDockWidget" name="asciidockWidget">
   <property name="minimumSize">
    <size>
     <width>200</width>
     <height>100</height>
    </size>
   </property>
   <property name="windowIcon">
    <iconset resource="myIcon.qrc">
     <normaloff>:/icon/image/icon/ASCII.png</normaloff>:/icon/image/icon/ASCII.png</iconset>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 10pt &quot;Elephant&quot;;</string>
   </property>
   <property name="windowTitle">
    <string>ASCII conversion</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_9">
    <widget class="QLabel" name="brightlabel_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#f8f8f8;&quot;&gt;Code value：&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>Input ASCII code</string>
     </property>
    </widget>
   </widget>
  </widget>
  <action name="action_N">
   <property name="text">
    <string>Open File(&amp;N)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_S">
   <property name="text">
    <string>Save(&amp;S)</string>
   </property>
  </action>
  <action name="action_H">
   <property name="text">
    <string>Svae Other(&amp;H)</string>
   </property>
  </action>
  <action name="action_Z">
   <property name="text">
    <string>Cancle(&amp;Z)</string>
   </property>
  </action>
  <action name="action_6">
   <property name="text">
    <string>Mean filtering</string>
   </property>
  </action>
  <action name="action_7">
   <property name="text">
    <string>Gray scale conversion</string>
   </property>
  </action>
  <action name="actionASCII">
   <property name="text">
    <string>ASCII conversion</string>
   </property>
  </action>
  <action name="action_Sobel">
   <property name="text">
    <string>Sobel edge extraction</string>
   </property>
  </action>
  <action name="action_Canny">
   <property name="text">
    <string>Canny edge detection</string>
   </property>
  </action>
  <action name="action_8">
   <property name="text">
    <string>Edge recombination
</string>
   </property>
  </action>
  <action name="action_9">
   <property name="text">
    <string>Binarization</string>
   </property>
  </action>
  <action name="action_10">
   <property name="text">
    <string>Gamma transformation</string>
   </property>
  </action>
  <action name="openfile">
   <property name="icon">
    <iconset resource="myIcon.qrc">
     <normaloff>:/icon/image/icon/open.png</normaloff>:/icon/image/icon/open.png</iconset>
   </property>
   <property name="text">
    <string>Open file</string>
   </property>
  </action>
  <action name="savafile">
   <property name="icon">
    <iconset resource="myIcon.qrc">
     <normaloff>:/icon/image/icon/sava.png</normaloff>:/icon/image/icon/sava.png</iconset>
   </property>
   <property name="text">
    <string>Save file</string>
   </property>
  </action>
  <action name="actionAbout_IMAGEW">
   <property name="text">
    <string>About Image</string>
   </property>
  </action>
  <action name="openfiles">
   <property name="icon">
    <iconset resource="myIcon.qrc">
     <normaloff>:/icon/image/icon/folder.png</normaloff>:/icon/image/icon/folder.png</iconset>
   </property>
   <property name="text">
    <string>Open files</string>
   </property>
  </action>
  <action name="action">
   <property name="text">
    <string>Gaussian filtering</string>
   </property>
  </action>
  <action name="action_3">
   <property name="text">
    <string>Open Files(&amp;P)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_4">
   <property name="text">
    <string>Brightness / contrast</string>
   </property>
  </action>
  <action name="action_ascii">
   <property name="icon">
    <iconset resource="myIcon.qrc">
     <normaloff>:/icon/image/icon/download.png</normaloff>:/icon/image/icon/download.png</iconset>
   </property>
   <property name="text">
    <string>Save ASCII</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>Local video</string>
   </property>
  </action>
  <action name="action_5">
   <property name="text">
    <string>Get camera</string>
   </property>
  </action>
  <action name="action_L">
   <property name="text">
    <string>Transtlater(&amp;L)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSmooth">
   <property name="text">
    <string>Smooth</string>
   </property>
  </action>
  <action name="actionMosaic">
   <property name="text">
    <string>Mosaic</string>
   </property>
  </action>
  <action name="actionBinarization">
   <property name="text">
    <string>Binarization</string>
   </property>
  </action>
  <action name="actionEdge_Detection">
   <property name="text">
    <string>Edge Detection</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="myIcon.qrc"/>
 </resources>
 <connections/>
</ui>
